samples/test/utils/kfp/samples/test/utils.py

```diff
@@ -257,16 +279,35 @@ def _create_run():
                             memory_limit='512Mi',
                         ))
                     if mode == kfp.dsl.PipelineExecutionMode.V1_LEGACY:
-                        conf.add_op_transformer(disable_cache)
-                    return client.create_run_from_pipeline_func(
-                        pipeline_func,
-                        pipeline_conf=conf,
-                        mode=mode,
-                        arguments=arguments,
-                        experiment_name=experiment,
-                        # This parameter only works for v2 compatible mode and v2 mode, it does not affect v1 mode
-                        enable_caching=enable_caching,
-                    )
+                        conf.add_op_transformer(_disable_cache)
+                    if pipeline_func:
+                        return client.create_run_from_pipeline_func(
+                            pipeline_func,
+                            pipeline_conf=conf,
+                            mode=mode,
+                            arguments=arguments,
+                            experiment_name=experiment,
+                        )
+                    else:
+                        pyfile = pipeline_file
+                        if pipeline_file.endswith(".ipynb"):
+                            pyfile = tempfile.mktemp(
+                                suffix='.py', prefix="pipeline_py_code")
+                            _nb_sample_to_py(pipeline_file, pyfile)
+                        from kfp.compiler.main import compile_pyfile
```

Comment:
I had to do this, because the `compile_pyfile` function is exactly what I want here.
Shall we consider making it part of the public API?